powerspace-model/src/main/protobuf/powerspace.proto

```diff
@@ -51,7 +51,18 @@ message BidExt {
     optional double ctr = 1000;
     optional double cpc = 1001;
     optional string algorithm = 1002;
-    optional double margin = 1003;
+
+    optional double advertiserCpm = 1006;
+    optional double advertiserCpc = 1004;
+
+    optional double finalRevenue = 1003; // == margin
+
+    optional double finalCpm = 1005;
+    optional double finalRevshare = 1007;
+    optional double internalDemandRevenue = 1008;
+
+    optional double publisherPayout = 1009;
+    optional double externalDemandRevenue = 1010;
```

Comment:
Do we really need all these new fields? If so, can you sort them by ID. Also, can you add update BidResponse JSON and tests?

Reply:
Will move it to private extensions. And I don't see interest in testing fully automatic code.